packages/quantic/force-app/main/default/lwc/quanticTimeframeFacet/quanticTimeframeFacet.js

```diff
@@ -0,0 +1,526 @@
+          })
+        : CoveoHeadless.buildDateRange({
+            start: {period: 'now'},
+            end: {
+              period: timeframe.period,
+              unit: timeframe.unit,
+              amount: timeframe.amount,
+            },
+          });
+    });
+  }
+
+  /**
+   * Gets the facet values with the formatting applied.
+   */
+  get formattedValues() {
+    const values = this.facetState?.values || [];
+
+    return values
+      .filter((value) => value.numberOfResults > 0)
+      .map((value) => ({
+        ...value,
+        label: this.formatFacetValue(value),
+        key: JSON.stringify([value.start, value.end]),
+        selected: value.state === 'selected',
+      }));
+  }
+
+  get actionButtonIcon() {
+    return this.isCollapsed ? 'utility:add' : 'utility:dash';
+  }
+
+  get actionButtonCssClasses() {
+    return this.isCollapsed ? 'facet__expand' : 'facet__collapse';
+  }
+
+  get actionButtonLabel() {
+    const label = this.isCollapsed
+      ? this.labels.expandFacet
+      : this.labels.collapseFacet;
+    return I18nUtils.format(label, this.label);
+  }
+
+  get hasActiveValues() {
+    return (
+      this.formattedValues.some((v) => v.selected) || this.dateFilterState.range
+    );
+  }
+
+  get clearFilterLabel() {
+    return I18nUtils.format(this.labels.clearFilter);
+  }
+
+  get datepickerFormat() {
+    return I18nUtils.getShortDatePattern();
+  }
+
+  /**
+   * @returns {DatepickerElement|null}
+   */
+  get startDatepicker() {
+    // @ts-ignore
+    return this.withDatePicker
+      ? this.template.querySelector('.timeframe-facet__start-input')
+      : null;
+  }
+
+  /**
+   * @returns {DatepickerElement|null}
+   */
+  get endDatepicker() {
+    // @ts-ignore
+    return this.withDatePicker
+      ? this.template.querySelector('.timeframe-facet__end-input')
+      : null;
+  }
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine 
+   */
+  initialize = (engine) => {
+    this.initializeSearchStatusController(engine);
+    this.initializeFacetController(engine);
+    this.initializeDateFilterController(engine);
+  };
+
+  /**
+   * @param {SearchEngine} engine 
+   */
+  initializeSearchStatusController(engine) {
+    this.searchStatus = CoveoHeadless.buildSearchStatus(engine);
+    this.unsubscribeSearchStatus = this.searchStatus.subscribe(() =>
+      this.updateSearchStatusState()
+    );
+  }
+
+  /**
+   * @param {SearchEngine} engine 
+   */
+  initializeFacetController(engine) {
+    this.facet = CoveoHeadless.buildDateFacet(engine, {
+      options: {
+        field: this.field,
+        currentValues: this.currentValues,
+        generateAutomaticRanges: false,
+        sortCriteria: 'descending',
+        filterFacetCount: !this.noFilterFacetCount,
+        injectionDepth: this.injectionDepth,
```

Comment:
```suggestion
        injectionDepth: Number(this.injectionDepth),
```
Otherwise it breaks when setting the option via markup.